spec/java/util/Random.main.lsl

```diff
@@ -63,6 +63,74 @@ automaton RandomAutomaton
 
     // utilities
 
+    @AutoInline @Phantom proc _throwIAE (): void
+    {
+        action THROW_NEW("java.lang.IllegalArgumentException", []);
+    }
+
+
+    @static proc _nextBytes (result: array<byte>, numBytes: int): void
+    {
+        val symbolicArray: array<byte> = action SYMBOLIC_ARRAY("byte", numBytes);
+        action ARRAY_COPY(symbolicArray, 0, result, 0, numBytes);
+    }
+
+
+    proc _generateRandomIntegerArrayWithBounds (size: int, randomNumberOrigin: int, randomNumberBound: int): array<int>
+    {
+        result = action SYMBOLIC_ARRAY("int", size);
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, size, +1,
+            checkIntBounds_loop(i, result, randomNumberOrigin, randomNumberBound)
+        );
+    }
+
+
+    @Phantom proc checkIntBounds_loop (i: int, result: array<int>, randomNumberOrigin: int, randomNumberBound: int): void
+    {
+        action ASSUME(result[i] >= randomNumberOrigin);
+        action ASSUME(result[i] < randomNumberBound);
+    }
+
+
+    proc _generateRandomLongArrayWithBounds (size: int, randomNumberOrigin: long, randomNumberBound: long): array<long>
+    {
+        result = action SYMBOLIC_ARRAY("long", size);
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, size, +1,
+            checkLongBounds_loop(i, result, randomNumberOrigin, randomNumberBound)
+        );
+    }
+
+
+    @Phantom proc checkLongBounds_loop (i: int, result: array<long>, randomNumberOrigin: long, randomNumberBound: long): void
+    {
+        action ASSUME(result[i] >= randomNumberOrigin);
+        action ASSUME(result[i] < randomNumberBound);
+    }
+
+
+    proc _generateRandomDoubleArrayWithBounds (size: int, randomNumberOrigin: double, randomNumberBound: double): array<double>
+    {
+        result = action SYMBOLIC_ARRAY("double", size);
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, size, +1,
+            checkDoubleBounds_loop(i, result, randomNumberOrigin, randomNumberBound)
+        );
+    }
+
+
+    @Phantom proc checkDoubleBounds_loop (i: int, result: array<double>, randomNumberOrigin: double, randomNumberBound: double): void
+    {
+        action ASSUME(result[i] != DOUBLE_NAN);
```

Comment:
Incorrect NaN check for Java translation target.
Please use this instead:
```dart
action ASSUME(result[i] == result[i]);
```

Also, consider making a local variable and then bind constraints on that:
```dart
val item: double = result[i];
action ASSUME(item == item);
...
```